layers/lispy/packages.el

```diff
@@ -0,0 +1,60 @@
+;;; packages.el --- lispy layer packages file for Spacemacs.
+;;
+;; Copyright (c) 2012-2016 Sylvain Benner & Contributors
+;;
+;; Author: Justin Burkett <justin@burkett.cc>
+;; URL: https://github.com/syl20bnr/spacemacs
+;;
+;; This file is not part of GNU Emacs.
+;;
+;;; License: GPLv3
+
+;;; Commentary:
+
+;;; Code:
+
+(defconst lispy-packages
+  '(lispy))
+
+(defun lispy/init-lispy ()
+  (use-package lispy
+    :config
+    (defun spacemacs/lispy-off ()
+      (when lispy-mode
+        (lispy-mode -1)))
+    (defvar-local spacemacs-lispy-mode nil)
+    (defun spacemacs/toggle-lispy ()
+      "Toggle `lispy-mode'. If `holy-mode' is enabled, this is
+just a simple toggle. For vim and hybrid style (depending on the
+value of `hybrid-mode') this adds or removes hooks that turn on
+`lispy-mode' only in hybrid/insert state and only in the current
+buffer."
+      (interactive)
+      (pcase (list (bound-and-true-p holy-mode)
+                   (bound-and-true-p hybrid-mode)
+                   spacemacs-lispy-mode)
+        (`(t ,x ,y)
+         (call-interactively 'lispy-mode))
+        (`(nil t t)
+         (message "Lispy-mode enabled in hybrid state for this buffer.")
+         (add-hook 'evil-hybrid-state-entry-hook 'lispy-mode nil t)
+         (add-hook 'evil-hybrid-state-exit-hook 'spacemacs/lispy-off nil t)
+         (setq spacemacs-lispy-mode nil))
+        (`(nil t nil)
+         (message "Lispy-mode disabled in hybrid state for this buffer.")
+         (remove-hook 'evil-hybrid-state-entry-hook 'lispy-mode t)
+         (remove-hook 'evil-hybrid-state-exit-hook 'spacemacs/lispy-off t)
+         (setq spacemacs-lispy-mode t))
+        (`(nil nil t)
+         (message "Lispy-mode enabled in insert state for this buffer.")
+         (add-hook 'evil-insert-state-entry-hook 'lispy-mode nil t)
+         (add-hook 'evil-insert-state-exit-hook 'spacemacs/lispy-off nil t)
+         (setq spacemacs-lispy-mode nil))
+        (`(nil nil nil)
+         (message "Lispy-mode disabled in insert state for this buffer.")
+         (remove-hook 'evil-insert-state-entry-hook 'lispy-mode t)
+         (remove-hook 'evil-insert-state-exit-hook 'spacemacs/lispy-off t)
+         (setq spacemacs-lispy-mode t))))
+    (spacemacs/set-leader-keys "k" 'spacemacs/toggle-lispy)))
```

Comment:
It is better to defer it if you can.
I cannot test the layer now, I have some papers to finish, I hope to be able to test it this evening.